browser/js/login/login.js

```diff
@@ -17,12 +17,20 @@ app.controller('LoginCtrl', function ($scope, AuthService, $state) {
 
         $scope.error = null;
 
-        AuthService.login(loginInfo).then(function () {
-            $state.go('home');
-        }).catch(function () {
-            $scope.error = 'Invalid login credentials.';
-        });
+        AuthService.login(loginInfo)
+            .then(function () {
+                $state.go('home');
+            })
+            .then(function () {
+                return CartFactory.getItems()
```

Comment:
Perhaps handle the cart initialization in the home state and let login concentrate on logging in.